Src/Sarif.PatternMatcher.Sdk/Crc32.cs

```diff
@@ -29,30 +37,46 @@ public static uint Calculate(byte[] buffer)
         }
 
         public static uint Calculate(uint checksum, byte[] buffer, int offset, int length)
+        {
+            return Calculate(checksum, buffer, offset, length, Crc32IEEETable);
+        }
+
+        public static uint Calculate(string text, uint[] crc32Table)
+        {
+            byte[] data = Encoding.ASCII.GetBytes(text);
+            return Calculate(data, crc32Table);
+        }
+
+        public static uint Calculate(byte[] buffer, uint[] crc32Table)
+        {
+            return Calculate(0, buffer, 0, buffer.Length, crc32Table);
+        }
```

Comment:
instead of creating a new set of functions, consider adding an optional parameter to existing functions, less code duplication.

Reply:
changed